osu.Game.Rulesets.Catch/Replays/CatchAutoGenerator.cs

```diff
@@ -44,9 +44,10 @@ void moveToNext(CatchHitObject h)
                 double timeAvailable = h.StartTime - lastTime;
 
                 //So we can either make it there without a dash or not.
-                double speedRequired = positionChange / timeAvailable;
+                double speedRequired = positionChange == 0 ? 0 : positionChange / timeAvailable;
```

Comment:
unnecessary, since 0 / timeAvailable = 0

Reply:
0/0=NaN

Reply:
The check should be on the second argument (denominator)

Reply:
Why exactly? If positionChange is non-zero and timeAvailable is zero, then speedRequired would be Infinity, as we want. If positionChange is zero, we never need to move so speedRequired should be zero.

Reply:
please document that this is expected directly above this line, as this is not apparent (and the changed code looks like a coding error).

Reply:
Updated the comment to specifically mention this. @smoogipoo worth double-checking this as to whether it is ambiguous in its current state.

Reply:
seems ok to me